### PR TITLE
[dxgi] Store device pointer in DxgiSwapChainDispatcher

### PR DESCRIPTION
**Pull request by gofman**
Helps REFramework.

### CODE REVIEW

File: src/dxgi/dxgi_swapchain_dispatcher.h [hunk 1]
```diff
@@ -296,6 +297,7 @@ namespace dxvk {
 
   private:
 
+    void* m_device_ptr;
```
Comment (Joshua-Ashton) [inline src/dxgi/dxgi_swapchain_dispatcher.h:300]: IUnknown* m_device;

Probably worth adding a comment as to why its not necessary to ref it either.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]

File: src/dxgi/dxgi_swapchain_dispatcher.h (hunk 1 above)
Comment (gofman) [reply src/dxgi/dxgi_swapchain_dispatcher.h:300]: Eh, that's exactly why I made it void * and suffixed _ptr, to suggest we have nothing to do with this data and specifically do not want to refcount it. Can be changed this way ofc.

Comment (Joshua-Ashton) [reply src/dxgi/dxgi_swapchain_dispatcher.h:300]: I will defer to @doitsujin's judgement then.

Comment (doitsujin) [reply src/dxgi/dxgi_swapchain_dispatcher.h]: Agree with Josh, we'd want that comment anyway to make it clear why we need an unused data member in the first place.

**Pull request merged**